src/macros.rs

```diff
@@ -470,19 +470,16 @@ macro_rules! crate_authors {
         impl Deref for CargoAuthors {
             type Target = str;
 
-            #[allow(unsafe_code)]
             fn deref(&self) -> &'static str {
                 static ONCE: Once = Once::new();
                 static mut VALUE: *const String = 0 as *const String;
 
-                unsafe {
-                    ONCE.call_once(|| {
-                        let s = env!("CARGO_PKG_AUTHORS").replace(':', $sep);
-                        VALUE = Box::into_raw(Box::new(s));
-                    });
+                ONCE.call_once(|| {
+                    let s = env!("CARGO_PKG_AUTHORS").replace(':', $sep);
+                    VALUE = Box::leak(s);
+                });
 
-                    &(*VALUE)[..]
-                }
+                &(*VALUE)[..]
```

Comment:
Surely this requires an `unsafe`, since it's dereferencing a raw pointer.

Perhaps this is a dumb question, but why can't we simply put an `Option<String>` on the `CargoAuthors` struct?

Reply:
Either `cargo check` lied to me, or the types and safety in here actually check out.

Reply:
This is a macro, which I'm pretty sure is not actually invoked inside clap, so just running a `check` is not sufficient.

Reply:
Oh! You're right. And tests indeed fail with type mismatch errors.

`Option<String>` sounds like a good idea as long as the `Option` is not visible in the public API.

Alternatively, we could just use `lazy_static!` instead of hand-rolling a replacement here.

Reply:
or use [OnceCell](https://docs.rs/once_cell/1.2.0/once_cell/)?

Reply:
I am not convinced that pulling in a dependency with a lot of `unsafe` internally is worthwhile. This code can be evaluated at compile time anyway, it should just be moved to compile time if it ever becomes a performance bottleneck.

Reply:
for some mysterious reason i was convinced that OneCell was free of `unsafe` so please forgive the interruption from the sidelines, and continue with the program…